### PR TITLE
Fix: fetch workflowpolicy policy info when request detail policy info

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/domain/service/application.go [hunk 1]
```diff
@@ -1199,17 +1204,19 @@ func (c *applicationServiceImpl) CreatePolicy(ctx context.Context, app *model.Ap
...
+	if !force {
+		used, err := c.checkPolicyUsedByWorkflow(ctx, app, policyName)
+		if err != nil {
+			return err
+		}
+		if used {
+			return bcode.ErrApplicationPolicyIsBeingUsed
+		}
 	}
 	if err := c.Store.Delete(ctx, &policy); err != nil {
```
[the remark this answers is not in the document]
Comment (wangyikewxgm) [reply pkg/apiserver/domain/service/application.go:1221]: Line 1228 doing this job.

File: pkg/apiserver/interfaces/api/application.go [hunk 2]
```diff
@@ -878,7 +879,9 @@ func (c *applicationAPIInterface) detailApplicationPolicy(req *restful.Request,
 
 func (c *applicationAPIInterface) deleteApplicationPolicy(req *restful.Request, res *restful.Response) {
 	app := req.Request.Context().Value(&apis.CtxKeyApplication).(*model.Application)
-	err := c.ApplicationService.DeletePolicy(req.Request.Context(), app, req.PathParameter("policyName"))
+	forceParam := req.QueryParameter("force")
```
Comment (barnettZQG) [inline pkg/apiserver/interfaces/api/application.go:882]: Create the parameter definition in the route. 

```
Param(ws.QueryParameter("force", "Force delete the policy and all references").DataType("boolean"))
```

File: pkg/apiserver/interfaces/api/dto/v1/types.go [hunk 3]
```diff
@@ -924,6 +925,8 @@ type PolicyBase struct {
 // DetailPolicyResponse app policy detail model
 type DetailPolicyResponse struct {
 	PolicyBase
+	// Binding relationShip
+	WorkflowPolicyBindings []WorkflowPolicyBinding `json:"workflowPolicyBind"`
```
Comment (barnettZQG) [inline pkg/apiserver/interfaces/api/dto/v1/types.go:929]:
```suggestion
	WorkflowPolicyBindings []WorkflowPolicyBinding `json:"workflowPolicyBind,omitempty"`
```

File: pkg/apiserver/domain/service/application.go [hunk 4]
```diff
@@ -1844,23 +1851,39 @@ func (c *applicationServiceImpl) handlePolicyBindingWorkflowStep(ctx context.Con
...
+	}
 	for _, w := range workflows {
+		var wlb apisv1.WorkflowPolicyBinding
+		have := false
 		for _, step := range w.Steps {
 			p := step.Properties
 			policies, _, err := extractPolicyListAndProperty(p)
 			if err != nil {
-				return false, err
+				return nil, err
```
Comment (StevenLeiZhang) [inline pkg/apiserver/domain/service/application.go:1874]: why is 'false' changed to be 'nil'?

File: pkg/apiserver/domain/service/application.go [hunk 5]
```diff
@@ -1844,23 +1851,39 @@ func (c *applicationServiceImpl) handlePolicyBindingWorkflowStep(ctx context.Con
...
 			}
 			for _, policy := range policies {
 				if policy == policyName {
-					return true, nil
+					wlb.Steps = append(wlb.Steps, step.Name)
+					have = true
```
Comment (StevenLeiZhang) [inline pkg/apiserver/domain/service/application.go:1879]: If  policy is found, break the iterator.

Comment (wangyikewxgm) [reply pkg/apiserver/domain/service/application.go:1879]: Cannot break, should find all workflow and workflowStep

File: pkg/apiserver/domain/service/application.go (hunk 4 above)
Comment (wangyikewxgm) [reply pkg/apiserver/domain/service/application.go:1874]: The function of this func has been changed to find all policy workflow bindings.

File: pkg/apiserver/domain/service/application_test.go [hunk 6]
```diff
@@ -936,35 +936,37 @@ var _ = Describe("Test apiserver policy rest api", func() {
...
-	It("Update workflow delete using step will unblock delete", func() {
+	It("Force delete policy will delete policy workflow binding", func() {
 		appModel, err := appService.GetApplication(context.TODO(), testApp)
 		Expect(err).Should(BeNil())
 		policyName := "override1"
-		policyRes, err := appService.DetailPolicy(ctx, appModel, policyName)
-		Expect(err).Should(BeNil())
-		propertyStr, err := json.Marshal(policyRes.Properties)
-		Expect(err).Should(BeNil())
-		updatePolicyReq := v1.UpdatePolicyRequest{
```
Comment (barnettZQG) [inline pkg/apiserver/domain/service/application_test.go:959]: Why remove the test case about the update.

Comment (wangyikewxgm) [reply pkg/apiserver/domain/service/application_test.go:959]: Force delete will do such things.

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**